core/src/limits.rs

```diff
@@ -0,0 +1,95 @@
+use crate::context::UpdateContext;
+use std::time::Duration;
+
+/// Indication of how long execution is allowed to take.
+///
+/// Execution is limited by two mechanisms:
+///
+/// 1. An *action limit*, which is a certain number of *actions* that can be
+///    taken before checking...
+/// 2. A *time limit*, which is checked every time the action limit runs out.
+///    If it has not yet expired, then the action limit is refreshed.
+///
+/// This two-tiered system is intended to reduce the overhead of enforcing
+/// execution limits.
+///
+/// What constitutes an "action" is up to the user of this structure. It may
+/// cover individual interpreter opcodes, bytes decoded in a data stream, or so
+/// on.
+pub struct ExecutionLimit {
+    /// How many actions remain before the next timelimit check.
+    ///
+    /// If `None`, then the execution limit is not enforced.
+    current_action_limit: Option<usize>,
```

Comment:
This is not the same as AVM1 actions, right? I saw it was used as equivalent to byte count at some other spot in code. Maybe some other name would be less ambiguous here.

Reply:
When I originally wrote this code I imagined it handling both the AVM and SWF execution limits, and I didn't call it `bytes` as I felt that would be too concrete. `actions` seemed generic enough to cover both counting AVM instructions and SWF tag bytes.

Of course, the aforementioned refactoring probably won't happen now so `bytes` might be fine.

Reply:
I'm renaming 'actions' to 'operations' here to remove the ambiguity